src/Command/ModelCommand.php

```diff
@@ -1546,4 +1549,41 @@ protected function bakeEnums(Table $model, array $data, Arguments $args, Console
             $enumCommand->execute($args, $io);
         }
     }
+
+    /**
+     * @param array<string, array<string, mixed>> $associations
+     * @return array<string, array<string, mixed>>
+     */
+    protected function ensureAliasUniqueness(array $associations): array
+    {
+        $existing = [];
+        foreach ($associations as $type => $associationsPerType) {
+            foreach ($associationsPerType as $k => $association) {
+                $alias = $association['alias'];
+                if (in_array($alias, $existing, true)) {
+                    $alias = $this->alias($association);
+                }
+                $existing[] = $alias;
+                if (empty($association['className'])) {
+                    $className = $this->plugin ? $this->plugin . '.' . $association['alias'] : $association['alias'];
+                    $association['className'] = $className;
+                }
+                $association['alias'] = $alias;
+                $associations[$type][$k] = $association;
+            }
+        }
+
+        return $associations;
+    }
+
+    /**
+     * @param array<string, mixed> $association
+     * @return string
+     */
+    protected function alias(array $association): string
```

Comment:
```suggestion
    protected function createAssociationAlias(array $association): string
```

`alias` is a pretty general term, I think a more specific name could be helpful for future us.